src/mlpack/methods/ann/activation_functions/softsign_function.hpp

```diff
@@ -77,24 +77,30 @@ class SoftsignFunction
   /**
    * Computes the first derivative of the softsign function.
    *
-   * @param y Input activation.
+   * @param x Input activation.
+   * @param y Result of Fn(x).
    * @return f'(x)
    */
-  static double Deriv(const double y)
+  static double Deriv(const double x, const double /* y */)
   {
-    return std::pow(1.0 - std::abs(y), 2);
+    // TODO: this was the old implementation of deriv, but I don't believe it is correct
+    // return std::pow(1.0 - std::abs(y), 2);
+    return 1.0 / std::pow(1.0 + std::abs(x), 2);
```

Comment:
Actually this is exactly the Elliot function... I would be in favor of removing one of the two.  Not sure which name is more widely used.  (Maybe we should open an issue about this and handle it later.)

Reply:
Yeh.  There's also the silu and swish functions, which are the same.  guess it depends on which one people usually use (or refer to it as).  or we could leave both headers and just make a typedef/alias so there's one implementation but people can call it whichever one they want.